compiler/src/lexer.mll

```diff
@@ -134,10 +136,16 @@ rule main = parse
   | '"' (([^'"' '\\']|'\\' _)* as s) '"' { STRING (unescape (L.of_lexbuf lexbuf) s) }
 
   (* Why this is needed *)
-  | ((*'-'?*) digit+) as s   
+  | ((*'-'?*) digit+('_'digit+)*) as s   
       {INT s} 
 
-  | ('0' ['x' 'X'] hexdigit+) as s
+  | ('0' ['x' 'X'] hexdigit+('_'hexdigit+)*) as s
+      {INT s}
+
+  | ('0' ['b' 'B'] bindigit+('_'bindigit+)*) as s
+      {INT s}
+
+  | ('0' ['o' 'O'] octdigit+('_'octdigit+)*) as s
```

Comment:
Why forbid consecutive underscores (as in `0xAA__BB`)?

Reply:
There was no reasons. I just didn't though of it at time. It is possible now